app/src/components/Sidebar.tsx

```diff
@@ -21,35 +21,62 @@ class Sidebar extends EnhancedComponent<ISidebarProps, ISidebarState> {
         {genre: "Rap", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
     ]
 
+    private shownGenres: ISidebarGenreChannel[] = [];
+
     protected constructor(props: ISidebarProps) {
         super(props);
         this.state = {
-            searchValue: "",
             genre: "",
             icon: "",
+            shownGenres: [
+                {genre: "Electronic", icon: "https://img.icons8.com/ios-glyphs/30/000000/electronic-music.png"},
+                {genre: "Rock", icon: "https://img.icons8.com/ios-glyphs/30/000000/rock-music.png"},
+                {genre: "Lo-Fi", icon: "https://img.icons8.com/ios-glyphs/30/000000/easy-listening.png"},
+                {genre: "Reggae", icon: "https://img.icons8.com/ios-glyphs/30/000000/reggae.png"},
+                {genre: "Country", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
+                {genre: "Hip-Hop", icon: "https://img.icons8.com/ios-glyphs/30/000000/hip-hop-music.png"},
+                {genre: "Jazz", icon: "https://img.icons8.com/ios-glyphs/30/000000/saxophone.png"},
+                {genre: "Rap", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
+            ],
         };
+        this.onSearch = this.onSearch.bind(this);
     }
 
     private onSearch(event: React.SyntheticEvent) {
+        event.preventDefault();
         let target = event.currentTarget as HTMLInputElement;
-        let value = target.value;
-        this.setState({
-            searchValue: value
-        });
+        let searchValue = target.getElementsByClassName("search_input")[0].getAttribute("value");
```

Comment:
typically with React, you should never need to do an HTML search like this. Do this with a ref; you can declare a variable in this class and set it in the render function. It'll look roughly something like this

```tsx
private searchBarRef: SearchBar;

constructor() {
  /**/
  this.saveSearchBarRef = this.saveSearchBarRef.bind(this);
}

private saveSearchBarRef(ref: SearchBar) {
    this.searchBarRef = ref;
}

render() {
  return(
    <SearchBar
        defaultText={/**/}
        // some other props here
        ref={this.saveSearchBarRef}
    />
  );
}
```

And now with the ref you can call SearchBar's functions with it, i.e. if you wrote a getter function for its text then you can do `this.searchBarRef.getText()`